### PR TITLE
Allow the creation of a(n almost) full geometry w/o a prior VMC instance

### PR DESCRIPTION
**Pull request by aphecetche**
Decoupling TGeo material creation from VMC (i.e. can now build TGeo material without having to instantiate first a TGeant3TGeo or TGeant4 object)

This is done by inverting the current responsabilities. TGeo is now the master in charge of
 creating the geometry, and VMC is just the client taking the geometry
 from TGeo (and possibly tweaking media properties which cannot be
 handled by TGeo itself).

The net advantage is to get a lighter dependency tree and is useful e.g. to build a geometry within a unit test.

### CODE REVIEW

**Comment by sawenzel** [comment]
Nice PR. I will check if the simulation still runs exactly as before by cross-checking the number of steps done in each module (using the StepLogger). If all goes fine, this will go in.

**Comment by sawenzel** [comment]
I am afraid that the first QA check did not pass. I ran the following command on dev and here:
```
DYLD_INSERT_LIBRARIES=libMCStepLogger.dylib o2sim -m FIT EMCAL TRD TPC ITS -n 1 -g pythia8
```

This is using a random selection of modules. I see a totally different number of steps done and much less secondaries for this new PR.

**Comment by aphecetche** [comment]
ok. will get back to the drawing board then ...

**Comment by aphecetche** [comment]
Hi @shahor02 

Indeed that must be the issue, and there's a massive difference because of the TPC SetSpecialPhysicsCuts call... (e.g. just using ITS or commenting out the SetSpecialPhysicsCuts calls, I get the same hit counts)

Have to think a bit ...

**Comment by shahor02** [comment]
@aphecetche : I think, also detectors which ConstructGeometry directly via ````TVirtualMC::GetMC()->Gsvolu```` (Calos, FIT,PHOS,TOF,TRD) should get problem, since they
use in ConstructGeometry media names which should be known to VMC.

Cheers,
 Ruben

**Comment by sawenzel** [comment]
Well for the moment you could just use in the MaterialManager (like in Detector): `if(vmc) { } else {
// use ROOT
}`

This should leave the simulation intact and allowing you to test geometry standalone.

**Comment by aphecetche** [comment]
@sawenzel 

indeed, after a bunch of inconclusive tests (meaning I still don't understand why I don't get the expected same results with TGeo driving the geometry creation), I added a couple of if in MaterialManager for the moment, to get the thing moving forward.

Create a JIRA ticket https://alice.its.cern.ch/jira/browse/AOCSF-135 for future work.

File: Detectors/Base/src/MaterialManager.cxx [hunk 1]
```diff
@@ -12,8 +12,8 @@
 /// \brief Implementation of the MaterialManager class
 
 #include "DetectorsBase/MaterialManager.h"
-#include <TVirtualMC.h>  // for TVirtualMC, gMC
-#include "TString.h"     // for TString
+#include "TVirtualMC.h"
```
Comment (sawenzel) [inline Detectors/Base/src/MaterialManager.cxx]: why these changes? They are not necessary for this PR, I guess.

Comment (aphecetche) [reply Detectors/Base/src/MaterialManager.cxx]: hum, no, indeed. In the original commit that include was no longer needed, then I put it back in the second commit but as a quotes include instead of angled include, out of habit I guess (as I've never quite understood why is Root considered a system include ;-) )

**Comment by sawenzel** [comment]
I squashed the 2 commits since it is not interesting to have the intermediate (breaking) state.
